libs/extensions/src/TTAlpsHistogramsFiller.cpp

```diff
@@ -112,4 +116,39 @@ void TTAlpsHistogramsFiller::FillHistograms1D(const std::shared_ptr<Event> event
       }
     }
   }
-}
\ No newline at end of file
+}
+
+void TTAlpsHistogramsFiller::FillLeadingPt(const std::shared_ptr<Event> event, std::string histName, std::vector<std::string> variableLocation) {
+  histogramsHandler->histograms1D[histName]->Fill(eventProcessor->GetMaxPt(event, variableLocation[0]));
+}
+
+void TTAlpsHistogramsFiller::FillSubLeadingPt(const std::shared_ptr<Event> event, std::string histName, std::vector<std::string> variableLocation) {
+  
+  float maxPt = eventProcessor->GetMaxPt(event, variableLocation[0]);
+  auto collection = event->GetCollection(variableLocation[0]);
+    for(auto object : *collection){
+      float pt = object->Get("pt");
+      if(pt == maxPt) continue;
+      histogramsHandler->histograms1D[histName]->Fill(pt);
+    }
```

Comment:
Doesn't this assume objects are sorted by pt? This is not always the case.

Reply:
Also, could be moved to EventProcessor, since GetMaxPt is already there.

Reply:
No this function fills all subleading pt (I changed the name to FillAllSubleadingPt now) so it will fill pt as long as pt != maxpt. I guess I could move it to EventProcessor, but then it would return a vector with all subleading pt